lib/bus_kiosk_web/live/kiosk_live.ex

```diff
@@ -117,6 +122,40 @@ defmodule BusKioskWeb.KioskLive do
     end
   end
 
+  defp setup_add_button_assigns(socket, stop_prediction_tuples) do
+    all_stops_have_predictions =
+      Enum.all?(stop_prediction_tuples, fn {_, _, predictions} ->
+        if length(predictions) > 0 do
+          true
+        else
+          false
+        end
+      end)
+
+    joined_stop_names =
+      Enum.map(stop_prediction_tuples, fn {_, name, _} -> name end)
+      |> Enum.join(",")
+
+    joined_stop_ids =
+      Enum.map(stop_prediction_tuples, fn {stop_id, _, _} -> stop_id end)
+      |> Enum.join(",")
+
+    save_stop_button_text =
+      if length(stop_prediction_tuples) > 1 do
+        "Save Stop Group"
+      else
+        "Save Stop"
+      end
+
+    if all_stops_have_predictions do
+      assign(socket, :joined_stop_names, joined_stop_names)
+      |> assign(:joined_stop_ids, joined_stop_ids)
+      |> assign(:save_stop_button_text, save_stop_button_text)
+    else
+      socket
+    end
+  end
+
```

Comment:
This is mostly related to wanting to minimize roundtrips. Moving it out of the view and into an assigns prevents that. It could be a LiveView component too, which would solve this in a more modular way too.

https://hexdocs.pm/phoenix_live_view/Phoenix.LiveView.html#module-change-tracking-pitfalls describes it better than me.